### ISSUE TITLE
command-line plugin doesn't recalculate width size for floats; forces horizontal scrollbars

### ISSUE THREAD

**Issue by garretwilson** [PrismJS/prism#965]
I have a something like this:

````html
<figure style="float:left; max-width: 50%;">
  <pre><code class="language-bash">cd foo/bar</code></pre>
<figure>
````

In other words, I have my Prism code section in a figure, and I float the entire figure on the left side, telling it that it can take up to 50% of the page. And it looks great! Obviously on most screens this figure is far smaller than the half the screen, so there is no scrollbars whatsoever.

But then if I turn on the command-line plugin, I have a problem:

````html
<figure style="float:left; max-width: 50%;">
  <pre class="command-line"><code class="language-bash">cd foo/bar</code></pre>
<figure>
````

The command-line plugin adds a left section with the *nix command prompt. Great---that's what I wanted. Unfortunately the size of the floated figure does not expand horizontally, even though there is plenty of room on the left half of the screen. Because the `<code>` section needs more room horizontally than the `<pre>` element gives it, the `<code>` section develops horizontal scrollbars. But they are unnecessary---the entire figure should have simply expanded horizontally!

In summary, the addition of the command-line plugin makes the `<code>` element require more room with the addition of the command prompt, but the outer `<pre>` element apparently still only reserves enough space for the `<code>` without the command prompt, resulting in unnecessary horizontal scrollbars.

**Comment by zeitgeist87** [comment]
Hi @garretwilson,

Thanks for reporting this! Could you provide a simple online example that demonstrates this behavior?

**Comment by garretwilson** [comment]
I'll provide you with complete HTML source code:

````html
<!DOCTYPE html>
<html lang="en-US" xmlns="http://www.w3.org/1999/xhtml">

<head>
	<meta charset="UTF-8" />
	<title>Prism Floating Command Line Bug</title>
	<link type="text/css" rel="stylesheet" href="prism.css" />
</head>

<body>
	<figure style="float:left; max-width: 50%;">
		<pre class="command-line"><code class="language-bash">mkdir doc
cd helloworld/src
javadoc -d ../doc com.example</code></pre>
	</figure>
	<script src="prism.js"></script>
</body>

</html>
````

To reproduce this issue you should download Prism from http://prismjs.com/download.html using the "coy" theme for _all_ languages, and include the following plugins:
* Line Highlight
* Line Numbers
* Keep Markup
* Command Line

**Comment by garretwilson** [comment]
Now my course is being translated to Portuguese, and we have a class using this next week. Still no movement on this? Anything I can do to get the ball rolling? Can I contribute some money somewhere to get someone to look at this? Thanks.

**Comment by zeitgeist87** [comment]
Hi @garretwilson,

Sorry for the late response, I was really busy in the last couple of months.

The `command-line` plugin uses the property `float: left` to position the command prompts. Unfortunately a float usually does not change the width of the parent. I've tried many things to solve this with floats and I came to the conclusion, that it is pretty much impossible.

But I found a working solution without `float` using `inline-blocks` [Branch](https://github.com/zeitgeist87/prism/tree/FixCommandLine) [Commit](https://github.com/zeitgeist87/prism/commit/6452356b03996adda34791d5d1abf6c659bcf32a). It is a bit of a hack. Maybe you could test it and if it works I would create a PR.

**Comment by zeitgeist87** [comment]
I found a bug in my solution. Here is the new [commit](https://github.com/zeitgeist87/prism/commit/eae5a4993ef8af625d738c4df6c9988769f56a96). It is quite possible, that this will break some stuff. We should test it extensively, before we apply it to the Prism.

**Comment by garretwilson** [comment]
Hi, @zeitgeist87. Thanks for looking into this. I've been so busy writing this course that I'm only now getting a chance to look back at the status here.

Can you clarify whether your commit got into a release, so I can try it out, or is it only on a branch? Or do I need to download your fork and try that?

**Comment by mAAdhaTTah** [comment]
@garretwilson I know it's been a while, but the commit referenced isn't in Prism, just his fork. You would need to copy the code in manually to your codebase to test if it resolves your issue. If it does, we can pull it in.

If you're not really having a problem anymore, we can close this issue. Let me know.

**Comment by garretwilson** [comment]
I'm still having this problem. In fact it cropped up just a few days ago. I had an example bash input in a figure that I had made be less than half the page horizontally. Once I turned on the command line plugin, even though the figure had plenty of space, it forced horizontal lines. If I remember correctly, it doesn't account for the extra command-line prompt that appears on the left. Unfortunately I don't have time to try pulling in the test fix right now. But it's not resolved, so it shouldn't be closed.
